test/gtest/ucp/test_ucp_peer_failure.cc

```diff
@@ -570,3 +621,163 @@ UCS_TEST_P(test_ucp_peer_failure_keepalive, kill_receiver,
+        add_variant_with_value(variants, UCP_FEATURE_TAG, TEST_TAG, "tag");
+    }
+
+    void protos_replace_progress()
+    {
+        for (int proto_id = 0; proto_id < ucp_protocols_count(); ++proto_id) {
+            auto proto = const_cast<ucp_proto_t*>(ucp_protocols[proto_id]);
+            m_progress_storage.emplace(proto, progress_array{});
+
+            int stage = UCP_PROTO_STAGE_START;
+            for (; stage < UCP_PROTO_STAGE_LAST; ++stage) {
+                m_progress_storage[proto][stage] = proto->progress[stage];
+                proto->progress[stage]           = &progress_wrapper;
+            }
+        }
+    }
+
+    void protos_restore_progress()
+    {
+        for (int proto_id = 0; proto_id < ucp_protocols_count(); ++proto_id) {
+            auto proto = const_cast<ucp_proto_t*>(ucp_protocols[proto_id]);
+
+            int stage = UCP_PROTO_STAGE_START;
+            for (; stage < UCP_PROTO_STAGE_LAST; ++stage) {
+                proto->progress[stage] = m_progress_storage[proto][stage];
+            }
+        }
+    }
+
+    void replace_ep_put_zcopy(ucp_ep_h ep)
+    {
+        for (auto lane = 0; lane < ucp_ep_num_lanes(ep); ++lane) {
+            if (lane == ucp_ep_get_cm_lane(ep)) {
+                continue;
+            }
+
+            auto iface  = ucp_ep_get_lane(ep, lane)->iface;
+            auto result = m_sender_uct_ops.emplace(iface, iface->ops);
+            if (result.second) {
+                iface->ops.ep_put_zcopy =
+                        reinterpret_cast<uct_ep_put_zcopy_func_t>(
+                                ucs_empty_function_return_unsupported);
+            }
+        }
+    }
+
+    static ucs_status_t progress_wrapper(uct_pending_req_t *uct_req)
+    {
+        auto req        = ucs_container_of(uct_req, ucp_request_t, send.uct);
+        auto proto      = req->send.proto_config->proto;
+        auto stage      = req->send.proto_stage;
+        auto proto_name = req->send.proto_config->proto->name;
+
+        if (std::string(proto_name) == "rndv/put/zcopy") {
+            if (stage == UCP_PROTO_STAGE_START) {
+                if (!self->m_is_ep_closed) {
+                    scoped_log_handler slh(wrap_errors_logger);
+                    self->receiver().close_all_eps(*self, 0,
+                                                   UCP_EP_CLOSE_FLAG_FORCE);
+                    self->replace_ep_put_zcopy(req->send.ep);
+                    self->m_is_ep_closed = true;
+                }
+            } else {
+                EXPECT_TRUE(false) << "Unexpected proto stage";
+            }
+        }
+
+        ucs_status_t status = self->m_progress_storage[proto][stage](uct_req);
+
+        if (self->m_is_ep_closed) {
+            for (auto &elem : self->m_sender_uct_ops) {
+                elem.first->ops = elem.second;
+            }
+            self->protos_restore_progress();
+        }
+
+        return status;
+    }
+
+    void rndv_put_abort_test()
```

Comment:
do we need this wrapper func? Looks like it is invoked by a single test `test_ucp_peer_failure_rndv_abort.put_zcopy`

Reply:
Removed.